Sources/SwiftKafka/KafkaProducer.swift

```diff
@@ -149,15 +147,7 @@ public final class KafkaProducer: Service, Sendable {
         let client = try RDKafka.createClient(
             type: .producer,
             configDictionary: config.dictionary,
-            deliveryReportCallback: { [logger, source] messageResult in
-                guard let messageResult else {
-                    logger.error("Could not resolve acknowledged message")
-                    return
-                }
-
-                // Ignore YieldResult as we don't support back pressure in KafkaProducer
-                _ = source.yield(messageResult)
-            },
+            events: RD_KAFKA_EVENT_LOG | RD_KAFKA_EVENT_DR,
```

Comment:
Can we abstract this into a Swift enum so we don't have to deal with C symbols here